lib/cache/mixins/resource/byUuid.js

```diff
@@ -0,0 +1,167 @@
+ | See the License for the specific language governing permissions and      |
+ |  limitations under the License.                                          |
+ ----------------------------------------------------------------------------
+
+  14 March 2019
+
+*/
+
+'use strict';
+
+const { logger } = require('../../../core');
+
+module.exports = (adapter) => {
+  return {
+
+    /**
+     * Checks if data resource by patient uuid or not
+     *
+     * @param  {string} resourceName
+     * @param  {string} uuid
+     * @return {bool}
+     */
+    exists: (resourceName, uuid) => {
+      logger.info('mixins/resource|byUuid|exists', { resourceName, uuid });
+
+      const key = ['Fhir', resourceName, 'by_uuid', uuid];
+
+      return adapter.exists(key);
+    },
+
+    /**
+     * Sets resource data
+     *
+     * @param  {string} resourceName
+     * @param  {string} uuid
+     * @param  {Object} resource
+     * @return {void}
+     */
+    set: (resourceName, uuid, resource) => {
+      logger.info('mixins/resource|byUuid|set', { resourceName, uuid, resource });
+
+      const key = ['Fhir', resourceName, 'by_uuid', uuid];
+      const dataKey = ['Fhir', resourceName, 'by_uuid', uuid, 'data'];
+
+      if (!adapter.exists(key)) {
+        adapter.putObject(dataKey, resource);
+      }
+    },
+
+    /**
+     * Gets resource data
+     *
+     * @param  {string} resourceName
+     * @param  {string} uuid
+     * @return {Object}
+     */
+    get: (resourceName, uuid) => {
+      logger.info('mixins/resource|byUuid|get', { uuid });
+
+      const key = ['Fhir', resourceName, 'by_uuid', uuid, 'data'];
+
+      return adapter.getObjectWithArrays(key);
+    },
+
+    /**
+     * Sets practitioner uuid for resource
+     *
+     * @param  {string} resourceName
+     * @param  {string} uuid
+     * @param  {string} practitionerUuid
+     * @return {void}
+     */
+    setPractitionerUuid: (resourceName, uuid, practitionerUuid) => {
+      logger.info('mixins/resource|byUuid|setPractitionerUuid', { resourceName, uuid, practitionerUuid });
+
+      const key = ['Fhir', resourceName, 'by_uuid', uuid, 'practitioner'];
+
+      adapter.put(key, practitionerUuid);
+    },
+
+    /**
+     * Gets practitioner uuid for resource
+     *
+     * @param  {string} resourceName
+     * @param  {string} uuid
+     * @return {string}
+     */
+    getPractitionerUuid: (resourceName, uuid) => {
+      logger.info('mixins/resource|byUuid|getPractitionerUuid', { resourceName, uuid });
+
+      const key = ['Fhir', resourceName, 'by_uuid', uuid, 'practitioner'];
+
+      return adapter.get(key);
+    },
+
+    /**
+     * Sets related uuid for resource
+     *
+     * @param  {string} resourceName
+     * @param  {string} uuid
+     * @param  {string} relatedResourceName
+     * @param  {string} relatedUuid
+     * @return {void}
+     */
+    setRelatedUuid: (resourceName, uuid, relatedResourceName, relatedUuid) => {
```

Comment:
@richardcbrown do we really need `toLowerCase()` here? I'd prefer to have it on other levels

Reply:
also, no test

Reply:
The toLowerCase here is really only to normalise the relatedResourceName that's passed in so users of the function don't have to worry about casing. If it was moved to a higher function it's be up to the developer to make sure they pass the right case to the get/set/exists, I'm not sure if it matters all that much though.